File: tests/solver_integration/cpp_solver/hello.cpp
// CoSimulation includes
#include "co_sim_io.hpp"

// move this macro to a checks.hpp in tests folder?
#define COSIMIO_CHECK_EQUAL(a, b)                                \
    if (a != b) {                                                \
        std::cout << "in line " << __LINE__ << " : " << a   \
                  << " is not equal to " << b << std::endl; \
        return 1;                                                \
    }
    
int main(){
    auto info = CoSimIO::Hello();

    std::cout << info << std::endl;

    int major_version = info.Get<int>("major_version");
    int minor_version = info.Get<int>("minor_version");
    std::string patch_version = info.Get<std::string>("patch_version");

    COSIMIO_CHECK_EQUAL(major_version, 0);
    COSIMIO_CHECK_EQUAL(minor_version, 0);


    return 0;
}

File: tests/solver_integration/python_solver/hello.py
import CoSimIO
info = CoSimIO.Hello()

def cosimio_check_equal(a, b):
    assert a is b


print(info)
major_version = info.GetInt("major_version")
minor_version = info.GetInt("minor_version")
patch_version = info.GetString("patch_version")

cosimio_check_equal(major_version,1)
cosimio_check_equal(minor_version,0)

print("CoSimIO version", str(major_version)+"."+str(minor_version)+"."+patch_version)